src/app/patron/books-borrowed-filter/books-borrowed-filter.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { BookBorrowedService } from 'src/app/service/book-borrowed.service';
+
+
+describe('BooksBorrowedFilterComponent', () => {
+    let comp: BooksBorrowedFilterComponent;
+    let fixture: ComponentFixture<BooksBorrowedFilterComponent>;
+    let de: DebugElement;
+    let el: HTMLElement;
+    let mockPatronService;
+    let mockBookBorrowedService;
+
+
+    beforeEach(fakeAsync(() => {
+
+        mockPatronService = jasmine.createSpyObj(['findBooksBorrowed','findBooksNotReturned']);
+        mockPatronService.findBooksBorrowed.and.returnValue(of([
+            {
+                id: 0, isbn: 'isbn1', dateBorrowed: '1', toBeReturned: '2', returnedOnTime: true, returned: true,
+                patron_id: 1
+            },
+            {
+                id: 2, isbn: 'isbn2', dateBorrowed: '1', toBeReturned: '2', returnedOnTime: false, returned: true,
+                patron_id: 1
+            }
+        ]
+        ));
+
+        mockPatronService.findBooksNotReturned.and.returnValue(of([
+            {
+                id: 0, isbn: 'isbn1', dateBorrowed: '1', toBeReturned: '2', returnedOnTime: true, returned: true,
+                patron_id: 1
+            }
+           
+        ]
+        ));
+
+        mockBookBorrowedService = jasmine.createSpyObj(['deleteBookBorrowed']);
+        mockBookBorrowedService.deleteBookBorrowed.and.returnValue(of([]));
+
+
+        TestBed.configureTestingModule({
+            declarations: [
+                BooksBorrowedFilterComponent
+            ],
+            imports: [
+                BrowserModule,
+                FormsModule,
+                ReactiveFormsModule,
+                RouterTestingModule,
+                HttpClientTestingModule
+            ],
+            providers: [
+                { provide: Router, useClass: class { navigate = jasmine.createSpy("navigate"); } },
+                { provide: AuthorService, useValue: [] },
+                {
+                    provide: ActivatedRoute, useValue: {
+                        snapshot: {
+                            params: {
+                                idPatron: 1
+                            },
+                        },
+                    },
+                },
+
+                { provide: PatronService, useValue: mockPatronService },
+                { provide: BookBorrowedService, useValue: mockBookBorrowedService }
+            ]
+        }).compileComponents().then(() => {
+            fixture = TestBed.createComponent(BooksBorrowedFilterComponent);
+            comp = fixture.componentInstance;
+            de = fixture.debugElement.query(By.css('form'));
+            el = de.nativeElement;
+            fixture.detectChanges();
+        });
+    }));
+
+    it('should create', () => {
+        expect(comp).toBeTruthy();
+    });
+
+    it(`should call the onSubmit method and set submitted to true`, fakeAsync(() => {
+      comp.onSubmit();
```

Comment:
Change the test title to reflect what the test actually does. ex ( should set submitted to true when onSubmit is called)